src/config.rs

```diff
@@ -203,8 +203,9 @@ pub struct Config {
     /// Specify the number of worker threads the Tokio Runtime will use.
     pub num_worker_threads: usize,
 
-    // If true, then use original source proxying
-    pub enable_original_source: Option<bool>,
+    // If set, explicitly configure whether to use original source.
+    // If unset (recommended), this is automatically detected based on permissions.
+    pub explicitly_configure_original_source: Option<bool>,
```

Comment:
```suggestion
     // If set, throw an error if we lack the permissions to enable original_source.
    // If unset (recommended), use of original_source is automatically detected based on permissions.
    pub require_original_source: Option<bool>,
```

maybe this?

Reply:
or maybe we just split this into two.

`require_original_source`
`disable_original_source`

Reply:
Or just a 3 way enum -

`orig_src: enable | disable | autodetect`

Reply:
> Or just a 3 way enum


If it wasn't clear, that is what `Option<bool>` is (just in a more confusing representation). Agree an enum is way more explicit.

Reply:
I guess there are two aspects to this:

1. internal representation, enum is awesome if we want to go to there, I have no problem changing from Option<bool> to something clearer internally.
2. external representation, enum is basically unsupported so it's string or int... I think keeping the optional bool and adding documentation if we feel it is unclear makes sense for this.